src/plugins/intel_gpu/src/graph/impls/ocl/kernel_selector_helper.cpp

```diff
@@ -1160,7 +1217,7 @@ void set_params(const kernel_impl_params& param_info, kernel_selector::params& p
     params.engineInfo.enable_sub_groups_emulation = true;
     params.engineInfo.bOptHintsSupport = false;
 
-    params.engineInfo.bLocalBlockIOSupport = device_info.supports_local_block_io && program->is_local_block_io_supported();
+    params.engineInfo.bLocalBlockIOSupport = query_local_block_io_supported(engine, config);
```

Comment:
If we query this here, we'll query every time of update_impl in dynamic shape which will affect the performance.. wont't it?

Reply:
I don't expect visible perf impact as starting from second set_params call result will be returned from cache, so it's just a matter of getting element from map